src/main/java/org/apache/maven/model/jdom/JDomDependencies.java

```diff
@@ -123,53 +121,26 @@ public Dependency set(int index, Dependency dependency) {
 
   @Override
   public void add(int index, Dependency dependency) {
-    addInternal(dependency, index);
-  }
+    if (index > size() || index < 0) {
+      throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size());
+    }
+
+    int elementIndex = index;
+    if (index > 0) {
+      Element previousElement = ((JDomDependency) get(index - 1)).getJDomElement();
+      elementIndex = 1 + getElementIndex(previousElement, jdomElement);
+    }
 
-  private boolean addInternal(Dependency dependency, int index) {
-    Element newElement;
+    JDomDependency jdomDependency;
     if (dependency instanceof JDomDependency) {
-      addElement(((JDomDependency) dependency).getJDomElement().clone(), jdomElement, index);
+      jdomDependency = (JDomDependency) dependency;
+      addElement(jdomDependency.getJDomElement().clone(), jdomElement, elementIndex);
     } else {
-      newElement = insertNewElement(POM_ELEMENT_DEPENDENCY, jdomElement, index);
-      JDomDependency jDomDependency = new JDomDependency(newElement);
-
-      jDomDependency.setGroupId(dependency.getGroupId());
-      jDomDependency.setArtifactId(dependency.getArtifactId());
-      jDomDependency.setVersion(dependency.getVersion());
-
-      String classifier = dependency.getClassifier();
-      if (classifier != null) {
-        jDomDependency.setClassifier(classifier);
-      }
-
-      List<Exclusion> exclusions = dependency.getExclusions();
-      if (!exclusions.isEmpty()) {
-        jDomDependency.setExclusions(exclusions);
-      }
-
-      if (dependency.isOptional()) {
-        jDomDependency.setOptional(true);
-      }
-
-      String scope = dependency.getScope();
-      if (!"compile".equals(scope)) {
-        jDomDependency.setScope(scope);
-      }
-
-
-      String systemPath = dependency.getSystemPath();
-      if (systemPath != null) {
-        jDomDependency.setSystemPath(systemPath);
-      }
-
-      String type = dependency.getType();
-      if (!"jar".equals(type)) {
-        jDomDependency.setType(type);
-      }
+      Element newElement = insertNewElement(POM_ELEMENT_DEPENDENCY, jdomElement, elementIndex);
+      jdomDependency = new JDomDependency(newElement, dependency);
     }
 
-    return super.add(dependency);
+    super.add(index, jdomDependency);
```

Comment:
Why don't you return the `super.add` result?

Reply:
Because the method is `void` (see next comment).